### ISSUE TITLE
Downloads on iOS don't complete

### ISSUE THREAD

**Issue by rdas1212** [jmshrv/finamp#433]
I'm on 0.6.11. Trying to download my playlists for offline play. For pretty much all my playlists, the app fails to download all the files. Some are stuck "enqueued" while others are in "running" for a long time. 
In the app logs I see logs like:
```
[DownloadsHelper] [<DateTime>] Download task list for com.unicornsonlsd.finamp-ios.download.task.<numbers>.<more numbers>.<some more numbers> (<some guid>) returned null, assuming item not downloaded.
```
These items don't show up when in offline mode, so I'm assuming the downloads are in fact failing. Any idea what's going on here?

**Comment by jmshrv** [comment]
Do you turn the phone screen off when downloading the playlist? Phones are weird and can kill the app when that happens.

**Comment by jmshrv** [comment]
Also that log message can be called from a few contexts, as it says it probably means the item didn't get downloaded

https://github.com/jmshrv/finamp/blob/194c1db14ef7a3a793b41299809ed1f0d5c785cd/lib/services/downloads_helper.dart#L572

**Comment by rdas1212** [comment]
I tried keeping the phone screen on & keeping the app in the foreground the whole time. Out of a 112 item playlist, 1 was stuck on running, 1 stuck enqueued. The next 268 item playlist got stuck with another 56 items running & 103 items enqueued.

**Comment by mikefictitious** [comment]
I am having the same issue on an iPhone 13 Pro running 16.4.1. Everything seemed fine on the previous version. On the latest version, if I click download on an album a bunch of songs will download immediately, but a random assortment will not. On the Downloads tab, they'll stay running or enqueued forever, but never fail. They won't play in Offline Mode.

Happy to provide whatever logs I can.

**Comment by jmshrv** [comment]
hmm, think I've managed to reproduce this. It looks like someone else has made another download package, maybe it'll be interesting for me to give it a go - https://pub.dev/packages/background_downloader

**Comment by pepega-h** [comment]
Hi, i just wanted to say that i am experiencing a similar issue on iPhone 13 16.3.1. 

Downloading a playlist misses a decent number of tracks but also makes Finamp extremely laggy until i remove all downloads. Sometimes the downloads dont even show in 'Offline mode'. When they do, offline mode is unusable because it can take 30 seconds before a song plays once i have pressed play. Right now i cant get them to play at all.

I have been pulling my hair out all day because i just migrated to a new Jellyfin install and had to delete and redownload my playlist so i thought that the migration could have caused the issue. I have deleted the app, rebooted phone, removed and re-downloaded several times etc.

I have tested on my iPad and that doesnt seem to have any issues.

I think the only relevant log file on the server is

[WRN] Slow HTTP Response from "http://192.168.1.99:8096/Items/***********/File" to "192.168.1.106" in 0:00:00.5570048 with Status Code 200

finamp is full of DownloadsHelper red logs 'download task list for com.unicornsonlsd.finamp-ios.download.task - returned null, assuming item not downloaded. stack trace null
